code/obj/item/swingsign.dm

```diff
@@ -0,0 +1,182 @@
+#define CHARACTERLIMIT 350
+
+//1. /obj/swingsign
+//	Inherited procs
+//	tgui procs
+//	procs
+//2. /obj/item/swingsignfolded
+
+TYPEINFO(/obj/swingsign)//No idea what TYPEINFO is, I just know it lets me disable the brown overcoat when crafting signs
+	mat_appearances_to_ignore = list("wood")
+/obj/swingsign
+	name = "swing sign"
+	desc = "A foldable sign for writing annoucements or advertisements."
+	icon = 'icons/obj/furniture/swingsign.dmi'
+	icon_state = "blank"
+	throwforce = 10
+	density = 1
+	anchored = UNANCHORED
+	/// Stored message
+	var/message = ""
+	var/defaultdesc = "A foldable sign for writing annoucements or advertisements."
+	/// There's a /div in SetMessage. If changing the preamble remember to respect it.
+	var/descpreamble = "It says:<br><div style='text-align:center'>"
+	var/secured = FALSE
+	/// Damage when thrown into a swing sign
+	var/maxcrashdamage = 5
+	/// Max length of the message
+	var/maxmessagerows = 10
+	/// Max width of the message
+	var/maxmessagecols = 40
+
+	proc/setmessage(var/newmessage)
+		message = newmessage
+		if(message == "")
+			desc = defaultdesc
+			icon_state = "blank"
+		else
+			desc = descpreamble + message + "</div>"
+			icon_state = "written"
+
+	proc/fold()
+		var/obj/item/swingsignfolded/C = new/obj/item/swingsignfolded(src.loc)
+
+		if (src.material)
+			C.setMaterial(src.material)
+		if (src.icon_state)
+			C.icon_state = "folded"
+		if (src.message)
+			C.message = src.message
+
+		qdel(src)
```

Comment:
not a big fan of deleting and spawning replacement items. Loses important metadata for admins as well as for players. Feel like this could be a single type instead of two, or possibly make use of a component similar to `/datum/component/foldable`